modules/systemd/host.nix

```diff
@@ -0,0 +1,345 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  # Ghaf configuration flags
+  cfg = config.ghaf.systemd.host;
+  cfgDebug = config.ghaf.profiles.debug.enable;
+  cfgGui = config.ghaf.graphics.weston.enable;
+
+  # Override minimal with host configuration set
+  packageConfiguration = pkgs.systemdMinimal.override {
+    pname = "host-systemd";
+    withAcl = true;
+    withAnalyze = cfg.withAnalyze;
+    withAudit = cfg.withAudit;
+    withCompression = cfgDebug || cfg.withVirtualization;
+    withCoredump = cfgDebug || cfg.withVirtualization;
+    withCryptsetup = cfg.withCryptsetup;
+    withEfi = pkgs.stdenv.hostPlatform.isEfi;
+    withFido2 = cfg.withFido;
+    withImportd = cfg.withVirtualization;
+    withKexectools = false;
+    withKmod = true;
+    withLibBPF = true;
+    withLibseccomp = true;
+    withLogind = cfg.withLogin;
+    withMachined = cfg.withVirtualization || cfg.withContainers;
+    withNetworkd = cfg.withNetwork;
+    withNss = cfg.withNameServiceSwitch;
+    withOomd = cfg.withOomd;
+    withPam = true;
+    withResolved = cfg.withNetwork || cfg.withNameServiceSwitch;
+    withSelinux = false;
+    withShellCompletions = cfgDebug;
+    withTimedated = true;
+    withTimesyncd = cfg.withNetwork;
+    withTpm2Tss = cfg.withTpm;
+    withUtmp = cfgDebug || (cfg.withAudit && cfg.withLogin);
+  };
+
+  # Definition of suppressed system units in systemd configuration. This removes the units and has priority.
+  # Required to avoid build failures when only disabling the units, and removes unit files in '/etc'. Note that
+  # errors will be silently ignored.
+  systemUnitConfiguration =
```

Comment:
I think `suppressedSystemUnits` would be more descriptive name. Then below you could just say `inherit suppressedSystemUnits`